client.go

```diff
@@ -345,6 +340,48 @@ func (uc *Client) isEnabled(feature string, options ...FeatureOption) api.Strate
 	}
 }
 
+func (uc *Client) isParentDependencySatisfied(feature *api.Feature, context context.Context) bool {
+    if feature == nil || feature.Dependencies == nil || len(*feature.Dependencies) == 0 {
+        return true
+    }
+
+    for _, parent := range *feature.Dependencies {
+        parentToggle := uc.repository.getToggle(parent.Feature)
+
+        if parentToggle == nil {
+            return false
+        }
+
+        if parentToggle.Dependencies != nil && len(*parentToggle.Dependencies) > 0 {
+            return false
+        }
+
+		// According to the schema, if the enabled property is absent we assume it's true.
+        if parent.Enabled == nil {
+            if parent.Variants != nil && len(*parent.Variants) > 0 {
+                variantName := uc.getVariantWithoutMetrics(parent.Feature, WithVariantContext(context)).Name
+                if contains(*parent.Variants, variantName) {
+                    continue
```

Comment:
not sure how idiomatic continue is in golang but I'd rather negate the contains and early exit with return false